src/test/java/com/exactpro/th2/crawler/CrawlerTest.java

```diff
@@ -187,4 +192,59 @@ public void dataServiceFail() throws IOException {
 
         Assertions.assertThrows(RuntimeException.class, crawler::process, exceptionMessage);
     }
+
+    @Test
+    @DisplayName("Crawler gets extra aliases while processing intervals")
+    public void dynamicAliasesProcessing() throws IOException, UnexpectedDataProcessorException {
+        CrawlerConfiguration configuration = CrawlerManager.createConfig(
+                "2021-06-16T12:00:00.00Z", DataType.MESSAGES, Set.of("alias\\d+"), SessionAliasType.REGEXP
+        );
+
+        CrawlerManager manager = new CrawlerManager(configuration);
+
+        List<Message> messages1 = MessageReaderKt.readMessages(Paths.get("src/test/resources/dynamic_aliases_messages-1.txt"));
+        List<Message> messages2 = MessageReaderKt.readMessages(Paths.get("src/test/resources/dynamic_aliases_messages-2.txt"));
+
+        List<String> aliases1 = new ArrayList<>(List.of("alias1", "alias2"));
+        StringList list1 = StringList.newBuilder().addAllListString(aliases1).build();
+
+        List<String> aliases2 = new ArrayList<>(List.of("alias1", "alias2", "alias3"));
+        StringList list2 = StringList.newBuilder().addAllListString(aliases2).build();
+
+        when(manager.getDataProviderMock().getMessageStreams(any(Empty.class))).thenReturn(list1, list2);
+
+        Crawler crawler = manager.createCrawler();
+
+        Iterator<StreamResponse> iterator1 = new MessageSearchResponse(messages1).iterator();
+        Iterator<StreamResponse> iterator2 = new MessageSearchResponse(messages2).iterator();
+
+        when(manager.getDataProviderMock().searchMessages(argThat(argument -> {
+            List<String> aliases = argument.getStream().getListStringList();
+
+            return aliases.containsAll(List.of("alias1", "alias2"));
+        }))).thenReturn(iterator1);
+
+        doReturn(iterator2).when(manager.getDataProviderMock()).searchMessages(argThat(argument -> {
+            List<String> aliases = argument.getStream().getListStringList();
+
+            return aliases.contains("alias3");
+        }));
+
+        when(manager.getDataServiceMock().sendMessage(any(MessageDataRequest.class))).thenReturn(MessageResponse.newBuilder().build());
+
+        crawler.process();
+
+        verify(manager.getIntervalsWorkerMock()).getIntervals(any(Instant.class), any(Instant.class), anyString(), anyString(), anyString());
+        verify(manager.getIntervalsWorkerMock()).storeInterval(any(Interval.class));
+
+        verify(manager.getDataProviderMock(), times(2)).searchMessages(any(MessageSearchRequest.class));
```

Comment:
Shouldn't we check the alias parameter of the `MessageSearchRequest` object?

Reply:
> I meant we should check the aliases retrieved from the data provider. The parameters verification should be done earlier in the constructor.

What kind of verification? In the constructor I already call method `matchRequestedSessionAliases()` that checks if `sessionAliasPatterns` is not null. If `sessionAliases` is empty, no request to the data-provider will be made due to this block:

`if (crawlerType == DataType.MESSAGES && parameters.getSessionAliases().isEmpty()) {
                        sendingReport = Report.empty();
                        break;
                    }`

Reply:
The `parameters.getSessionAliases()` might not be empty but the provided simply does not have aliases that match the specified parameters

Reply:
As far as I can see, `parameters.getSessionAliases()` will never contain aliases that do not match the specified parameters. In the constructor those aliases are set with `new HashSet<>(matchRequestedSessionAliases())` that will check the aliases before returning them. Further, the aliases from `parameters.getSessionAliases()` are set with `getNewAliases()` method that calls `matchRequestedSessionAliases()` internally, too.

Reply:
I think we don't understand each other. I mean that the actual session aliases retrieved from the data provider might be empty (`new HashSet<>(matchRequestedSessionAliases())`). And we should check them instead `parameters.getSessionAliases()`. If the aliases retrieved from the data provider are empty there is no point in making a request to get messages/events.

Reply:
I think I got your point. I made a commit where I check `sessionAliases` before making any requests.